src/planner.cc

```diff
@@ -556,13 +556,16 @@ void Planner::Impl::bring_path_to_halt() {
 }
 
 void Planner::Impl::GetCurrentPosition(AxesRegister *pos) {
-  // TODO: this should use the MotorOperations::GetPhysicalStatus to get
-  // the very last postition where we are in.
-  assert(planning_buffer_.size() > 0);  // we always should have a current pos
-  const int *mpos = planning_buffer_[0]->position_steps;
+  PhysicalStatus physical_status;
+  motor_ops_->GetPhysicalStatus(&physical_status);
+  int axis_steps;
   for (const GCodeParserAxis a : AllAxes()) {
-    if (cfg_->steps_per_mm[a] != 0) {
-      (*pos)[a] = 1.0f * mpos[a] / cfg_->steps_per_mm[a];
+    const bool axis_mapped =
+      hardware_mapping_->AssignAxisSteps(a, physical_status.pos_steps, &axis_steps);
```

Comment:
With the GetAxisSteps() this loop can be simplified to
```
 pos->zero();
  for (const GCodeParserAxis a : AllAxes()) {
      if (cfg_->steps_per_mm[a] != 0) {
              (*pos)[a] = GetAxisSteps(a, physical_status) / cfg_->steps_per_mm[a]);
      }
  }
```

Reply:
(Solved)